### ISSUE TITLE
Hi, does the google colab support training dreambooth on SDXL?

### ISSUE THREAD

**Issue by MaxTran96** [TheLastBen/fast-stable-diffusion#2338]
Hi i have a few questions
1st, does the google colab fast-stable diffusion support training dreambooth on SDXL?
2nd, I see there's a train_dreambooth.py and train_dreambooth_lora.py. What's the difference between them? i also see there's a train_dreambooth_lora_sdxl.py, when will there be a pure dreambooth version of sdxl? i.e train_dreambooth_sdxl.py. Another question is, is it possible to pass negative prompt into SDXL? Thank you so much Ben!

**Comment by MaxTran96** [comment]
The one on google colab fast stable diffusion is using full dreambooth or dreambooth lora? Whats the difference between full dreambooth and dreambooth lora? Sorry if its a silly question

**Comment by TheLastBen** [comment]
> The one on google colab fast stable diffusion is using full dreambooth or dreambooth lora? Whats the difference between full dreambooth and dreambooth lora? Sorry if its a silly question

LoRA is a smaller model less than 1GB that contains the trained concept, can be plugged-in in A1111, not as good as full dreambooth, but in the SDXL case, it's good enough

**Comment by MaxTran96** [comment]
what do you mean by full dreambooth here? The actual dreambooth model? From what i understand, dreambooth is a model and it can be combined with various finetune techinque such as Lora, am i correct? What do you mean by full dreambooth here. What finetune technique does the fast-stable-diffusion google colab use? Full dreambooth?

**Comment by TheLastBen** [comment]
colab notebook offers full dreambooth for v1.5 and v2.1, meaning the output model is the full 2gb model, a LoRA is no more than 200mb.

**Comment by MaxTran96** [comment]
when will sdxl training feature be added on google colab?

**Comment by TheLastBen** [comment]
> when will sdxl training feature be added on google colab?

probably won't be added to colab, google is contently putting obstacles to such notebooks, so at the moment it's avaliable on Runpod and Paperspace only

**Comment by MaxTran96** [comment]
i'm building a web application for my side project, is it possible to know the code to run training and inference for sdxl model? Thanks!

**Comment by TheLastBen** [comment]
https://github.com/TheLastBen/diffusers/blob/main/examples/dreambooth/train_dreambooth_rnpd_sdxl_lora.py

**Comment by MaxTran96** [comment]
I have a quick question, in the google colab fast-stable-diffusion, it doesn't seem to generate regularization images, i thought dreambooth requires regularization image to effectively train your face. Did you intentionally leave it out?

**Comment by TheLastBen** [comment]
My dreambooth method is different, it doesn't use regularization, and the instance prompt is just one single unknown token.

**Comment by MaxTran96** [comment]
Quick question, what's the difference between train_dreambooth_rnpd_sdxl_lora.py and train_dreambooth_sdxl_lora.py? Which one should i use?

**Comment by MaxTran96** [comment]
I think the only differences are vae.to(accelerator.device, dtype=torch.float32) ->  vae.to(accelerator.device, dtype=weight_dtype)
and 
batch["pixel_values"]=(vae.encode(batch["pixel_values"].to(accelerator.device, dtype=weight_dtype)).latent_dist.sample() * vae.config.scaling_factor)

**Comment by TheLastBen** [comment]
they are the same, just tiny changes

**Comment by MaxTran96** [comment]
Did you update the fast dreambooth jupyter notebook? I now ran into issue of python version. Do you know why this happened? Also do you have the code to run inference for sdxl model?


    self.process_distribution(spec, dist, deps)
  File "/usr/lib/python3/dist-packages/setuptools/command/easy_install.py", line 784, in process_distribution
    distros = WorkingSet([]).resolve(
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 781, in resolve
    dist = best[req.key] = env.best_match(
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 1066, in best_match
    return self.obtain(req, installer)
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 1078, in obtain
    return installer(requirement)
  File "/usr/lib/python3/dist-packages/setuptools/command/easy_install.py", line 712, in easy_install
    return self.install_item(spec, dist.location, tmpdir, deps)
  File "/usr/lib/python3/dist-packages/setuptools/command/easy_install.py", line 738, in install_item
    dists = self.install_eggs(spec, download, tmpdir)
  File "/usr/lib/python3/dist-packages/setuptools/command/easy_install.py", line 923, in install_eggs
    return self.build_and_install(setup_script, setup_base)
  File "/usr/lib/python3/dist-packages/setuptools/command/easy_install.py", line 1191, in build_and_install
    self.run_setup(setup_script, setup_base, args)
  File "/usr/lib/python3/dist-packages/setuptools/command/easy_install.py", line 1177, in run_setup
    run_setup(setup_script, args)
  File "/usr/lib/python3/dist-packages/setuptools/sandbox.py", line 253, in run_setup
    raise
  File "/usr/lib/python3.8/contextlib.py", line 131, in __exit__
    self.gen.throw(type, value, traceback)
  File "/usr/lib/python3/dist-packages/setuptools/sandbox.py", line 195, in setup_context
    yield
  File "/usr/lib/python3.8/contextlib.py", line 131, in __exit__
    self.gen.throw(type, value, traceback)
  File "/usr/lib/python3/dist-packages/setuptools/sandbox.py", line 166, in save_modules
    saved_exc.resume()
  File "/usr/lib/python3/dist-packages/setuptools/sandbox.py", line 141, in resume
    six.reraise(type, exc, self._tb)
  File "/usr/lib/python3/dist-packages/setuptools/_vendor/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "/usr/lib/python3/dist-packages/setuptools/sandbox.py", line 154, in save_modules
    yield saved
  File "/usr/lib/python3/dist-packages/setuptools/sandbox.py", line 195, in setup_context
    yield
  File "/usr/lib/python3/dist-packages/setuptools/sandbox.py", line 250, in run_setup
    _execfile(setup_script, ns)
  File "/usr/lib/python3/dist-packages/setuptools/sandbox.py", line 45, in _execfile
    exec(code, globals, locals)
  File "/tmp/easy_install-n4cy5z6p/numpy-1.25.1/setup.py", line 22, in <module>
    
RuntimeError: Python version >= 3.9 required.

**Comment by MaxTran96** [comment]
i got the error from running  python3 setup.py install (which is python 3.8)

**Comment by MaxTran96** [comment]
i fixed that issue, by using the older version of transformers and numpy. Another issue i have is i got this error 

Traceback (most recent call last):
  File "/home/ubuntu/content/diffusers/examples/dreambooth/train_dreambooth_rnpd_sdxl_lora.py", line 782, in <module>
    main()
  File "/home/ubuntu/content/diffusers/examples/dreambooth/train_dreambooth_rnpd_sdxl_lora.py", line 777, in main
    network.save_weights(model_path, torch.float32, None)
  File "/home/ubuntu/content/diffusers/examples/dreambooth/lora_sdxl.py", line 1005, in save_weights
    torch.save(state_dict, file)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/torch/serialization.py", line 440, in save
    with _open_zipfile_writer(f) as opened_zipfile:
  File "/home/ubuntu/.local/lib/python3.8/site-packages/torch/serialization.py", line 315, in _open_zipfile_writer
    return container(name_or_buffer)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/torch/serialization.py", line 288, in __init__
    super().__init__(torch._C.PyTorchFileWriter(str(name)))
RuntimeError: [enforce fail at inline_container.cc:365] . invalid file name: .safetensors

this is my command
accelerate launch /home/ubuntu/content/diffusers/examples/dreambooth/train_dreambooth_rnpd_sdxl_lora.py\
    --image_captions_filename \
    --train_text_encoder \
    --pretrained_model_name_or_path="stabilityai/stable-diffusion-xl-base-0.9" \
    --instance_data_dir=/home/ubuntu/content/VenusAI/Avatar/Sessions/test/instance_images \
    --output_dir=/home/ubuntu/content/VenusAI/Avatar/Sessions/test/output \
    --captions_dir=/home/ubuntu/content/VenusAI/Avatar/Sessions/test/captain \
    --instance_prompt="" \
    --seed=123456 \
    --resolution=1024 \
    --mixed_precision=fp16 \
    --train_batch_size=1 \
    --gradient_accumulation_steps=1 --gradient_checkpointing \
    --use_8bit_adam \
    --learning_rate=2e-6 \
    --lr_scheduler="linear" \
    --lr_warmup_steps=0 \
    --max_train_steps=80 
    
    i thought you can pass in the safetensor file no?

**Comment by MaxTran96** [comment]
sorry if i'm asking too much, but how do i use convert_original_stable_diffusion_to_diffusers.py to convert .safetensor sdxl file to diffusers so that i can pass the output into pretrained_model_name_or_path to run

accelerate launch /home/ubuntu/content/diffusers/examples/dreambooth/train_dreambooth_rnpd_sdxl_lora.py\
    --image_captions_filename \
    --train_text_encoder \
    --pretrained_model_name_or_path="stabilityai/stable-diffusion-xl-base-0.9" \ 
    --instance_data_dir=/home/ubuntu/content/VenusAI/Avatar/Sessions/test/instance_images \
    --output_dir=/home/ubuntu/content/VenusAI/Avatar/Sessions/test/output \
    --captions_dir=/home/ubuntu/content/VenusAI/Avatar/Sessions/test/captain \
    --instance_prompt="" \
    --seed=123456 \
    --resolution=1024 \
    --mixed_precision=fp16 \
    --train_batch_size=1 \
    --gradient_accumulation_steps=1 --gradient_checkpointing \
    --use_8bit_adam \
    --learning_rate=2e-6 \
    --lr_scheduler="linear" \
    --lr_warmup_steps=0 \
    --max_train_steps=80

**Comment by mahaboobkhan29** [comment]
I'm a beginner programmer and have spent the past few days browsing stackoverflow and the relevant communities, but couldn't find the solution to my problem. Would appreciate if anyone has any insight to the following.

I am trying to train my dreambooth using the colab: https://colab.research.google.com/github/ShivamShrirao/diffusers/blob/main/examples/dreambooth/DreamBooth_Stable_Diffusion.ipynb

But when tried to train on stabilityai/stable-diffusion-xl-base-1.0 and default settings, I encountered:

Revision Not Found for url: https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0/resolve/fp16/model_index.json. Invalid rev id: fp16

So I removed fp16 and then encountered:

**AttributeError: 'UNet2DConditionModel' object has no attribute 'add_embedding'. Did you mean: 'class_embedding'?**

I tried various solutions like updating my python and changing to a different diffuser version, but still encountered the error above. It seems i'd have to dig into the source code, but as a beginner programmer I have barely any idea what I'm doing. Again, would appreciate if anyone has any idea or point me to the correct repo.
